platforms/software/dependency-management/src/main/java/org/gradle/internal/component/model/DefaultExternalComponentGraphResolveState.java

```diff
@@ -231,4 +237,38 @@ public AttributesSchemaInternal getAttributesSchema() {
             return metadata.getAttributesSchema();
         }
     }
+
+    private static class ExternalGraphSelectionCandidates implements GraphSelectionCandidates {
```

Comment:
Copied from `AbstractComponentGraphResolveState.java`, but with the explicit check for `isCanBeConsumed` replaced with an assert since all external components always return consumable configurations.